internal/lib/sandbox/builder.go

```diff
@@ -0,0 +1,393 @@
+	SetConfig(*types.PodSandboxConfig) error
+
+	// GenerateNameAndID sets the sandbox name and ID
+	GenerateNameAndID() error
+
+	// Config returns the sandbox configuration
+	Config() *types.PodSandboxConfig
+
+	// ID returns the id of the pod sandbox
+	ID() string
+
+	// Name returns the id of the pod sandbox
+	Name() string
+
+	// InitInfraContainer initializes the sandbox's infra container
+	InitInfraContainer(*libconfig.Config, *storage.ContainerInfo, *idtools.IDMappings) error
+
+	// Spec returns the infra container's generator.
+	// Must be called after InitInfraContainer.
+	Spec() *generate.Generator
+
+	// ResolvPath returns the sandbox's resolvPath.
+	ResolvPath() string
+
+	// SetDNSConfig sets the dns configs.
+	SetDNSConfig(*types.DNSConfig)
+
+	// SetCRISandbox sets and creates CriSandbox.
+	SetCRISandbox(string, map[string]string, map[string]string, *types.PodSandboxMetadata) error
+
+	// GetSandbox gets the sandbox and deletes the config and sandbox.
+	GetSandbox() (*Sandbox, error)
+
+	// SetNamespace sets the namespace.
+	SetNamespace(string)
+
+	// SetName sets the name for the sandbox
+	SetName(string)
+
+	// SetKubeName sets the kubename.
+	SetKubeName(string)
+
+	// SetLogDir sets the logDir of the sandbox
+	SetLogDir(string)
+
+	// SetContainers sets the containers.
+	SetContainers(memorystore.Storer[*oci.Container])
+
+	// SetProcessLabel sets the processLabel.
+	SetProcessLabel(string)
+
+	// SetMountLabel sets the mountLabel.
+	SetMountLabel(string)
+
+	// SetShmPath sets the shim path.
+	SetShmPath(string)
+
+	// SetCgroupParent sets the cgroup parent.
+	SetCgroupParent(string)
+
+	// SetPrivileged sets the privileged.
+	SetPrivileged(bool)
+
+	// SetRuntimeHandler sets the runtime handler.
+	SetRuntimeHandler(string)
+
+	// SetResolvPath sets the resolv.conf path.
+	SetResolvPath(string)
+
+	// SetHostname sets the hostname.
+	SetHostname(string)
+
+	// SetPortMappings sets the port mappings.
+	SetPortMappings([]*hostport.PortMapping)
+
+	// SetHostNetwork sets the host network.
+	SetHostNetwork(bool)
+
+	// SetUsernsMode sets the user namespace mode.
+	SetUsernsMode(string)
+
+	// SetPodLinuxOverhead sets the PodLinuxOverhead.
+	SetPodLinuxOverhead(*types.LinuxContainerResources)
+
+	// SetPodLinuxResources sets the PodLinuxResources.
+	SetPodLinuxResources(*types.LinuxContainerResources)
+
+	// SetHostnamePath sets the hostname path.
+	SetHostnamePath(string)
+
+	// SetNamespaceOptions sets the namespace options.
+	SetNamespaceOptions(*types.NamespaceOption)
+
+	// SetSeccompProfilePath sets the selinux comp profile path.
+	SetSeccompProfilePath(string)
+
+	// SetID sets the Id inside the criSandbox object or creates it.
+	SetID(string)
+
+	// SetCreatedAt sets the created at time.
+	SetCreatedAt(createdAt time.Time)
+}
+
+// builder is the hidden default type behind the Sandbox interface.
+type builder struct {
+	config *types.PodSandboxConfig
+	infra  container.Container
+	sb     *Sandbox
```

Comment:
nit: Can we rename this to `sandboxRef` or `sandbox`?

Reply:
Changed to sandboxRef